From 7f9bfec9bdc13f387565ddeaa48cc4c093654867 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@freescale.com>
Date: Mon, 13 Jul 2015 16:28:22 +0800
Subject: [PATCH] MLK-11242-1 video: mxsfb: delay set_par actions when fb is
 in blank

When the fb is in blank mode, do set_par has no real meaning.
So delaying this action in the later unblank request.

Signed-off-by: Fancy Fang <chen.fang@freescale.com>
---
 drivers/video/mxsfb.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/video/mxsfb.c b/drivers/video/mxsfb.c
index e9c61df..e14847b 100644
--- a/drivers/video/mxsfb.c
+++ b/drivers/video/mxsfb.c
@@ -665,6 +665,14 @@ static int mxsfb_set_par(struct fb_info *fb_info)
 	clk_enable_pix(host);
 
 	dev_dbg(&host->pdev->dev, "%s\n", __func__);
+
+	/* If fb is in blank mode, it is
+	 * unnecessary to really set par here.
+	 * It can be delayed when unblank fb
+	 */
+	if (host->cur_blank != FB_BLANK_UNBLANK)
+		return 0;
+
 	/*
 	 * It seems, you can't re-program the controller if it is still running.
 	 * This may lead into shifted pictures (FIFO issue?).
-- 
1.8.0

